src/structs/from_iter.rs

```diff
@@ -0,0 +1,42 @@
+use crate::{Generator, GeneratorResult, ValueResult};
+
+/// Creates a generator that wraps an `Iterator`.
+///
+/// # Examples
+///
+/// Basic usage:
+///
+/// ```
+/// use crate::pushgen::GeneratorExt;
+/// let v = vec![1, 2, 3];
+/// let mut gen = pushgen::from_iter(v);
+///
+/// let mut output: Vec<i32> = Vec::new();
+/// gen.for_each(|x| output.push(x));
+/// assert_eq!(output, [1, 2, 3]);
+/// ```
+pub fn from_iter<I: IntoIterator>(iterable: I) -> FromIter<I::IntoIter> {
+    FromIter(iterable.into_iter())
+}
+
+/// A generator where each iteration delegates to an `Iterator`
+///
+/// This `struct` is created by the [`from_iter()`] function.
+/// See its documentation for more.
+///
+/// [`from_iter()`]: crate::from_iter
+#[derive(Clone)]
+pub struct FromIter<I>(I);
+
+impl<I: Iterator> Generator for FromIter<I> {
+    type Output = I::Item;
+
+    fn run(&mut self, mut output: impl FnMut(Self::Output) -> ValueResult) -> GeneratorResult {
```

Comment:
Could you add `#[inline]` on this?

Should `run` be implemented in terms of `Iterator::try_for_each`?

Reply:
I think to get the early return working with `try_for_each` would require a `impl Try` which can't be done yet.  If you know of a way, feel free to point it put.

Reply:
Wouldn't it be possible to return an `Option::None ` for the early return case? But that might be more hassle than necessary? We can leave that for now